packages/vector_graphics/lib/src/listener.dart

```diff
@@ -561,6 +569,14 @@ class FlutterVectorGraphicsListener extends VectorGraphicsCodecListener {
 
   @override
   void onSize(double width, double height) {
+    final double targetWidth = _targetSize?.width ?? width;
+    final double targetHeight = _targetSize?.height ?? height;
+
+    final double sx = targetWidth / width;
+    final double sy = targetHeight / height;
+
+    _canvas.scale(min(sx, sy));
```

Comment:
https://api.flutter.dev/flutter/dart-ui/Canvas/scale.html

You need to use separate x/y scaling otherwise this will won't work correctly for non-uniform scales